### ISSUE TITLE
Cannot restore sysNAND backup on 2.1.0 2ds (v0.83)

### ISSUE THREAD

**Issue by zXTechnoWolf** [d0k3/OTPHelper#9]
Latest version, nothing seems to be working. Made sure the backups were 943 MB (988,807,168 bytes), everything should be fine. The backups I made for rednand and sysnand were not mixed up, and OTPHelper is reporting incorrect file size when I try to restore. Followed Plailect's guide with an o3ds and it worked flawlessly.

**Comment by Neocube96** [comment]
Similar problem on version 0.83. When restoring sysNAND_original.bin it gives an error and says there is no usable file available. Worked though when using the previous version of OTPHelper.
On o3ds xl

**Comment by LITTOMA** [comment]
Same problem on 0.83 on old3ds.

**Comment by lowbraincount** [comment]
I also have the same problem on N3ds xl I've just obtained my OTP and am trying to restore from sysNAND_original.bin yet it claims its the wrong size (despite me checking that it was the correct size 1.21gb) Can I recover from Sysnand.bin (which is for some reason 1.84gb)

**Comment by TheDgtl** [comment]
Issue happens with o3ds as well. Have two friends following the guide right now, only occurred on the o3ds with a redNAND, not the one with a standard emuNAND

**Comment by vqvu** [comment]
Seems to be a regression from 8fb7c946f866589e6165f51513635569c70390e6. You now get a min-sized sysNAND dump if you are using a min-sized redNAND. Problem is, the validation code doesn't know this, so if your NAND chip is larger than the min-size, it ends up thinking that you have a backup that's too small.

I applied this patch to get going again.

```diff
diff --git a/source/decryptor/nand.c b/source/decryptor/nand.c
index 57a15e7..5f0d94f 100644
--- a/source/decryptor/nand.c
+++ b/source/decryptor/nand.c
@@ -139,6 +139,11 @@ u32 CheckNandIntegrity(const char* path)
     if (path) {
         if (!DebugFileOpen(path))
             return 1;
+
+        if (emunand_offset + getMMCDevice(0)->total_size > NumHiddenSectors()) {
+            nand_size = NAND_MIN_SIZE;
+        }
+
         if (nand_size != FileGetSize()) {
             FileClose();
             Debug("NAND backup has the wrong size!");
```

**Comment by d0k3** [comment]
Everything should be fixed in v0.84, so I'm closing this. @vqvu, yup, that was one of the problems. @lowbraincount both NAND backups are valid, the smaller one just only uses the actual used size of the NAND (ie the minimum size). In general, if a NAND backup passes OTPHelpers checking and OTPHelper restores it, it is fine.
